src/core/gltf2exporter/dracocompressor_p.h

```diff
@@ -0,0 +1,69 @@
+/*
+    dracocompressor_p.h
+
+    This file is part of Kuesa.
+
+    Copyright (C) 2019 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com
+    Author: Juan Jose Casafranca <juan.casafranca@kdab.com>
+
+    Licensees holding valid proprietary KDAB Kuesa licenses may use this file in
+    accordance with the Kuesa Enterprise License Agreement provided with the Software in the
+    LICENSE.KUESA.ENTERPRISE file.
+
+    Contact info@kdab.com if any conditions of this licensing are not clear to you.
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Affero General Public License as
+    published by the Free Software Foundation, either version 3 of the
+    License, or (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef KUESA_GLTF2EXPORTER_DRACOCOMPRESSOR_H
+#define KUESA_GLTF2EXPORTER_DRACOCOMPRESSOR_H
+
+//
+//  NOTICE
+//  ------
+//
+// We mean it: this file is not part of the public API and could be
+// modified without notice
+//
+
+#include <memory>
+#include <vector>
+#include <QString>
+
+namespace draco {
+class EncoderBuffer;
+}
+
+QT_BEGIN_NAMESPACE
+
+namespace Qt3DRender {
+class QGeometry;
+}
+namespace Kuesa {
+class GLTF2ExportConfiguration;
+namespace DracoCompression {
+
+struct CompressedMesh {
+    ~CompressedMesh();
```

Comment:
why is the destructor needed? Adding it disables the generation of the default copy constructor which (with clang) leads to compile errors

Reply:
weird, if I remove the destructor it builds for me. Stupid compilers.

Don't mind on the fix, but if you add the destructor then you need also add the copy constructor (at least, maybe the regular constructor too) for clang to be happy

Reply:
> weird, if I remove the destructor it builds for me. Stupid compilers.

I just checked and this looks like it's a matter of language version used - this does not work for me on gcc with C++11 either

Reply:
ah, the reason why it may have worked without destructor nor include, is because all the files that use dracocompressor_p.h also include the required draco header - but a new source file that would just do 

     #include "dracocompressor_p.h"
     void foo() { CompressedMesh mesh; } 

would fail to build